isthmus/src/main/java/io/substrait/isthmus/SqlToSubstrait.java

```diff
@@ -49,18 +53,41 @@
 /** Take a SQL statement and a set of table definitions and return a substrait plan. */
 public class SqlToSubstrait {
 
+  public Plan execute(String sql, Function<List<String>, NamedStruct> tableLookup)
+      throws SqlParseException {
+    RelDataTypeFactory factory = new JavaTypeFactoryImpl();
+    Function<List<String>, Table> lookup =
+        id -> {
+          NamedStruct table = tableLookup.apply(id);
+          if (table == null) {
+            return null;
+          }
+          List<RelDataType> types =
+              table.struct().fields().stream()
+                  .map(f -> TypeConverter.convert(factory, f))
```

Comment:
I don't think this will work correctly for nested structs, will it? The names list is a tree of names and we'd need to somehow pass that into TypeConverter.

Reply:
yup, good catch. Fixed